pymc/distributions/shape_utils.py

```diff
@@ -471,19 +471,18 @@ def convert_size(size: Size) -> Optional[StrongSize]:
     """Process a user-provided size variable into None or a valid size object."""
     if size is None:
         return None
-
-    if isinstance(size, int) or (isinstance(size, TensorVariable) and size.ndim == 0):
-        size = (size,)
+    if isinstance(size, int):
+        size = size
+    elif isinstance(size,TensorVariable):
```

Comment:
Please check for the ndim too (like a few lines above). For ndim>1 we still want to run into the exception case.